modules/l4remoteiplist/matcher.go

```diff
@@ -0,0 +1,126 @@
+// Copyright (c) 2024 SICK AG
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package l4remoteiplist
+
+import (
+	"fmt"
+	"net"
+	"net/netip"
+
+	"github.com/caddyserver/caddy/v2"
+	"github.com/caddyserver/caddy/v2/caddyconfig/caddyfile"
+
+	"github.com/mholt/caddy-l4/layer4"
+	"go.uber.org/zap"
+)
+
+func init() {
+	caddy.RegisterModule(&RemoteIPList{})
+}
+
+type RemoteIPList struct {
+	RemoteIPFile string `json:"remote_ip_file"`
+
+	logger       *zap.Logger
+	remoteIPList *IPList
+}
+
+// CaddyModule returns the Caddy module information.
+func (*RemoteIPList) CaddyModule() caddy.ModuleInfo {
+	return caddy.ModuleInfo{
+		ID:  "layer4.matchers.remote_ip_list",
+		New: func() caddy.Module { return new(RemoteIPList) },
+	}
+}
+
+// Provision implements caddy.Provisioner.
+func (m *RemoteIPList) Provision(ctx caddy.Context) error {
+	m.logger = ctx.Logger()
+
+	remoteIPList, err := NewIPList(m.RemoteIPFile, &ctx, m.logger)
```

Comment:
```suggestion
	remoteIPList, err := NewIPList(m.RemoteIPFile, ctx, m.logger)
```

Passing a pointer to a context is probably a bug...